rflx/pyrflx/typevalue.py

```diff
@@ -725,27 +765,14 @@ def set_refinement(fld: MessageValue.Field, fld_name: str) -> None:
             raise KeyError(f"cannot access field {field_name}")
 
         self.__update_simplified_mapping()
+        check_outgoing_condition_satisfied()
 
-        if all(
-            [
-                self.__simplified(o.condition) == FALSE
-                for o in self._type.outgoing(Field(field_name))
-            ]
-        ):
-            self._fields[field_name].typeval.clear()
-            if isinstance(value, bytes):
-                value_repr = "x" + value.hex()
-            else:
-                value_repr = str(value)
-
-            raise ValueError(
-                f"none of the field conditions "
-                f"{[str(o.condition) for o in self._type.outgoing(Field(field_name))]}"
-                f" for field {field_name} have been met by the assigned value: {value_repr}"
-            )
-
-        self._preset_fields(field_name)
-        self._update_accessible_fields()
+        if not prevent_recursive_checksum_calc:
+            self._preset_fields(field_name)
+            self._update_accessible_fields()
+            for checksum_aspect in self._checksums.values():
```

Comment:
`self._checksums` contains no aspects: Rename `checksum_aspect` to `checksum`.

Reply:
Changed in c238ac0